src/app/core/models/customer/customer.mapper.ts

```diff
@@ -50,4 +50,11 @@ export class CustomerMapper {
           isBusinessCustomer: false,
         };
   }
+
+  private static isBusinessCustomer(data: CustomerData): boolean {
+    if (data.type === 'PrivateCustomer') {
```

Comment:
As discussed, the type information is almost irrelevant. It just tells you what REST resource you were just using. It should not be used for distinction.
The marker for distinction should be the returned `customerType`. This is not yet implemented (see IS-30018) so as a work around the return of a `companyName` property is used as marker for now to distinct BusinessCustomers (has a companyName) from PrivateCustomers.

Reply:
This condition is right, the distinction is made by the question which rest resource is used
The service decides which rest resource it asks, so mapper doesn't have to take care of this